tjpcov/covariance_clusters.py

```diff
@@ -374,3 +350,41 @@ def double_bessel_integral(self, z1, z2):
             float: Numerical approximation of integral.
         """
         return self.fft_helper.two_fast_algorithm(z1, z2)
+
+    def _build_matrix_from_blocks(self, blocks, tracers_cov):
+        """Build full matrix from blocks.  Cluster covariance overrides this
+        because our tracer combinations are not unique per data type.
+
+        Args:
+            blocks (list): List of blocks
+            tracers_cov (list): List of tracer combinations corresponding to
+                each block in blocks. They must have the same order
+
+        Returns:
+            array: Covariance matrix for all combinations in the sacc file.
+        """
+        blocks = iter(blocks)
+
+        s = self.io.get_sacc_file()
+        ndim = s.mean.size
+
+        cov_full = np.nan * np.ones((ndim, ndim))
+
+        print("Building the covariance: placing blocks in their place")
+        for tracer_comb1, tracer_comb2 in tracers_cov:
+            # We do not need to do the reshape here because the blocks have
+            # been build looping over the data types present in the sacc file
+            print(tracer_comb1, tracer_comb2)
+
+            cov_ij = next(blocks)
+            ix1 = s.indices(
+                data_type=standard_types.cluster_counts, tracers=tracer_comb1
```

Comment:
Spoke in slack.  Basically right now we are unsure about where/how the covariance for the cluster weak lensing mass measurements will be calculated.  As it stands it is misleading to name our class "Cluster Covariance" since that alludes to including all cluster measurements.  To avoid confusion I renamed the base class to ClusterCountsCovariance so we are explicit about only calculating the covariance for cluster counts.  This does not address the issue of having no covariance calculated for some of our data vector (cluster weak lensing mass measurements).  Will discuss with other members of the cluster group to figure out what to do here!